src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,45 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserRegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    public static final int MIN_LENGTH_LOGIN_PARAMETER = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserExist(user);
+        checkUserLoginLength(user);
+        checkUserPassLength(user);
+        checkUserAge(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUserExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserRegisterException("User exists!");
+        }
+    }
+
+    private void checkUserLoginLength(User user) {
+        if (user.getLogin().length() < MIN_LENGTH_LOGIN_PARAMETER) {
+            throw new UserRegisterException("Login is short!");
+        }
+    }
+
+    private void checkUserPassLength(User user) {
+        if (user.getPassword().length() < MIN_LENGTH_LOGIN_PARAMETER) {
```

Comment:
Bug: You are checking the password length against the constant `MIN_LENGTH_LOGIN_PARAMETER`. It might be better to introduce a new constant for the password length to make the code more readable and avoid potential errors. For example, `MIN_LENGTH_PASSWORD_PARAMETER`.

Reply:
Done